### PR TITLE
interp: do not wrap empty interface

### LINKED ISSUE #829: interp: cannot type assert empty interface populated by json into slice
The following program `panic.go` triggers a panic:

```go
package main

import (
	"encoding/json"
	"fmt"
	"log"
)

const jsonData = `[
  {"foo": "foo"},
  {"bar": "bar"}
]`

func main() {
	var a interface{}

	if err := json.Unmarshal([]byte(jsonData), &a); err != nil {
		log.Fatalln(err)
	}

	fmt.Printf("%T\n", a)
	b := a.([]interface{})

	for _, v := range b {
		fmt.Println(v)
	}
}

```

Expected result:
```console
$ go run ./panic.go
[]interface {}
map[foo:foo]
map[bar:bar]
```

Got:
```console
$ yaegi ./panic.go
panic.go:15:6: panic
reflect.Set: value of type *interp.valueInterface is not assignable to type *interface {}
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc000370600, 0x40)
	/usr/lib/go/src/runtime/debug/stack.go:24 +0x9f
github.com/containous/yaegi/interp.(*Interpreter).eval.func1(0xc00028bcb0)
	/home/sh0rez/work/yaegi/interp/interp.go:362 +0xc5
panic(0xda4fe0, 0xc000114e60)
	/usr/lib/go/src/runtime/panic.go:969 +0x175
github.com/containous/yaegi/interp.runCfg.func1(0xc0003ea0a0, 0xc000127600)
	/home/sh0rez/work/yaegi/interp/run.go:123 +0x217
panic(0xda4fe0, 0xc000114e60)
	/usr/lib/go/src/runtime/panic.go:969 +0x175
reflect.Value.assignTo(0xd86c60, 0xc00011d4c0, 0x16, 0xed8d6d, 0xb, 0xd88ce0, 0x0, 0x4d0a56, 0xe25040, 0xd86c60)
	/usr/lib/go/src/reflect/value.go:2425 +0x405
reflect.Value.Set(0xd88ce0, 0xc00011a248, 0x196, 0xd86c60, 0xc00011d4c0, 0x16)
	/usr/lib/go/src/reflect/value.go:1554 +0xbd
github.com/containous/yaegi/interp.addr.func1(0xc0003ea0a0, 0xc00011d300)
	/home/sh0rez/work/yaegi/interp/run.go:523 +0xba
github.com/containous/yaegi/interp.runCfg(0xc000127600, 0xc0003ea0a0)
	/home/sh0rez/work/yaegi/interp/run.go:129 +0x6e
github.com/containous/yaegi/interp.(*Interpreter).run(0xc00013e1a0, 0xc000126e00, 0xc0003ea000)
	/home/sh0rez/work/yaegi/interp/run.go:107 +0x2af
github.com/containous/yaegi/interp.(*Interpreter).eval(0xc00013e1a0, 0xc000368420, 0x148, 0x7ffff85db470, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/sh0rez/work/yaegi/interp/interp.go:454 +0x75c
github.com/containous/yaegi/interp.(*Interpreter).EvalPath(0xc00013e1a0, 0x7ffff85db470, 0x8, 0xc000368200, 0x148, 0x0, 0x0, 0xc00028bd68)
	/home/sh0rez/work/yaegi/interp/interp.go:346 +0xd5
main.runFile(0xc00013e1a0, 0x7ffff85db470, 0x8, 0xc00010e180, 0x1)
	/home/sh0rez/work/yaegi/cmd/yaegi/run.go:122 +0x1c8
main.run(0xc00000e090, 0x1, 0x1, 0xf12938, 0xdb6860)
	/home/sh0rez/work/yaegi/cmd/yaegi/run.go:80 +0x7cc
main.main()
	/home/sh0rez/work/yaegi/cmd/yaegi/yaegi.go:128 +0x29b
```

### LINKED ISSUE #984: interp: panic when getting address of empty interface
The following program `sample.go` triggers a panic:

```go
package main

func main() {
	var a interface{}

	println(&a)
}
```

Expected result:
```console
$ go run ./sample.go
0xc00003c768
```

Got:
```console
$ yaegi ./sample.go
./sample.go:6:10: panic
run: reflect: call of reflect.Value.Interface on zero Value
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0002d4600, 0x40)
	/Users/mpl/go1/src/runtime/debug/stack.go:24 +0x9f
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0xc000115c70)
	/Users/mpl/src/github.com/traefik/yaegi/interp/interp.go:480 +0xc5
panic(0x19f5340, 0xc00000ee20)
	/Users/mpl/go1/src/runtime/panic.go:969 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc00035a0a0, 0xc000456a00, 0xc000115a60)
	/Users/mpl/src/github.com/traefik/yaegi/interp/run.go:184 +0x247
panic(0x19f5340, 0xc00000ee20)
	/Users/mpl/go1/src/runtime/panic.go:969 +0x1b9
reflect.valueInterface(0x0, 0x0, 0x0, 0x101, 0x1a43620, 0xc00000ee00)
	/Users/mpl/go1/src/reflect/value.go:1020 +0x1ad
reflect.Value.Interface(...)
	/Users/mpl/go1/src/reflect/value.go:1015
github.com/traefik/yaegi/interp.addr.func1(0xc00035a0a0, 0xc00042c990)
	/Users/mpl/src/github.com/traefik/yaegi/interp/run.go:682 +0xf9
github.com/traefik/yaegi/interp.runCfg(0xc000456a00, 0xc00035a0a0)
	/Users/mpl/src/github.com/traefik/yaegi/interp/run.go:190 +0x87
github.com/traefik/yaegi/interp.(*Interpreter).run(0xc000354000, 0xc000456200, 0xc00035a000)
	/Users/mpl/src/github.com/traefik/yaegi/interp/run.go:121 +0x2af
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc000354000, 0xc00003c7c0, 0x40, 0x7ffeefbff6da, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/mpl/src/github.com/traefik/yaegi/interp/interp.go:572 +0x809
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc000354000, 0x7ffeefbff6da, 0xb, 0xc00003c700, 0x40, 0x0, 0x0, 0xc000116f70)
	/Users/mpl/src/github.com/traefik/yaegi/interp/interp.go:394 +0xfa
main.runFile(0xc000354000, 0x7ffeefbff6da, 0xb, 0xc00007e300, 0x1)
	/Users/mpl/src/github.com/traefik/yaegi/cmd/yaegi/run.go:123 +0xae
main.run(0xc0000a21a0, 0x1, 0x1, 0x13, 0x13)
	/Users/mpl/src/github.com/traefik/yaegi/cmd/yaegi/run.go:89 +0x7e5
main.main()
	/Users/mpl/src/github.com/traefik/yaegi/cmd/yaegi/yaegi.go:133 +0x479
```

### LINKED ISSUE #1015: interface was invalid memory address or nil pointer dereference when type check
The following program `sample.go` triggers a panic:

```go
package main

import (
	"encoding/json"
	"fmt"
	"net/url"
)

func main() {
	body := []byte(`{
		"BODY_1": "VALUE_1",
		"BODY_2": "VALUE_2",
		"BODY_3": null,
		"BODY_4": {
			"BODY_1": "VALUE_1",
			"BODY_2": "VALUE_2",
			"BODY_3": null
		},
		"BODY_5": [
			"VALUE_1",
			"VALUE_2",
			"VALUE_3"
		]
	}`)

	values := url.Values{}

	var rawData map[string]interface{}
	err := json.Unmarshal(body, &rawData)
	if err != nil {
		fmt.Println("can't parese body")
	}

	for key, val := range rawData {
		switch val.(type) {
		case string, bool, float64:
			values.Add(key, fmt.Sprint(val))
		case nil:
			values.Add(key, "")
		case map[string]interface{}, []interface{}:
			jsonVal, err := json.Marshal(val)
			if err != nil {
				fmt.Println("can't encode json")
			}
			values.Add(key, string(jsonVal))
		}
	}
	fmt.Println(values.Get("BODY_1"))
	fmt.Println(values.Get("BODY_2"))
	fmt.Println(values.Get("BODY_3"))
	fmt.Println(values.Get("BODY_4"))
	fmt.Println(values.Get("BODY_5"))
}

```

Expected result:
```console
$ go run ./sample.go
VALUE_1
VALUE_2

{"BODY_1":"VALUE_1","BODY_2":"VALUE_2","BODY_3":null}
["VALUE_1","VALUE_2","VALUE_3"]
```

Got:
```console
$ yaegi ./sample.go
run: runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0002d8800, 0x40)
	/usr/local/Cellar/go/1.15.6/libexec/src/runtime/debug/stack.go:24 +0x9f
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0xc0000c1c70)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/interp.go:485 +0xc5
panic(0x1a0ade0, 0x2126d50)
	/usr/local/Cellar/go/1.15.6/libexec/src/runtime/panic.go:969 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc000356210, 0xc000117400, 0xc0000c1a60)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/run.go:184 +0x247
panic(0x1a0ade0, 0x2126d50)
	/usr/local/Cellar/go/1.15.6/libexec/src/runtime/panic.go:969 +0x1b9
github.com/traefik/yaegi/interp.(*itype).id(0x0, 0xc00047e4e0, 0x199)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/type.go:1082 +0x3a
github.com/traefik/yaegi/interp._case.func5(0xc000356210, 0xc00007f5c0)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/run.go:2644 +0xfc
github.com/traefik/yaegi/interp.runCfg(0xc000117400, 0xc000356210)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/run.go:190 +0x87
github.com/traefik/yaegi/interp.(*Interpreter).run(0xc0000321c0, 0xc000116c00, 0xc000356000)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/run.go:121 +0x2b0
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc0000321c0, 0xc00045a400, 0x3d8, 0x7ffeefbffb9e, 0x9, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/interp.go:577 +0x809
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc0000321c0, 0x7ffeefbffb9e, 0x9, 0xc00045a000, 0x3d8, 0x0, 0x0, 0xc000110ea0)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/interp/interp.go:399 +0xfa
main.runFile(0xc0000321c0, 0x7ffeefbffb9e, 0x9, 0xc0000403c0, 0x1)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/cmd/yaegi/run.go:123 +0xae
main.run(0xc000190030, 0x1, 0x1, 0x13, 0x13)
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/cmd/yaegi/run.go:89 +0x7e5
main.main()
	/Users/olaf/go/pkg/mod/github.com/traefik/yaegi@v0.9.11/cmd/yaegi/yaegi.go:144 +0x415
```

### PR DESCRIPTION
**Pull request by mpl**
The empty interface (interface{}), and its variants (such as []interface{} and map[string]interface{}), are commonly used in Go to (json) Unmarshal arbitrary data. Within Yaegi, all interface types are wrapped in a valueInterface struct in order to retain all the information needed for a consistent internal state (as reflect is not enough to achieve that). However, this wrapping ends up being problematic when it comes to the type assertions related to the aforementioned Unmarshaling.

Therefore, this PR is an attempt to consider the empty interface (and its variants) as an exception within Yaegi, that should never be wrapped within a valueInterface, and to treat it similarly to the other basic Go types. The assumption is that the wrapping should not be needed, as there is no information about implemented methods to maintain.

Fixes #984 
Fixes #829 
Fixes #1015

### CODE REVIEW

**Review by soulbalz — APPROVED** [review]

**Review by soulbalz — APPROVED** [review]

File: internal/cmd/genop/genop.go [hunk 1]
```diff
@@ -63,23 +63,73 @@ func {{$name}}(n *node) {
...
+			if n.typ.cat != interfaceT || len(n.typ.field) > 0 {
+				n.exec = func(f *frame) bltn {
+					_, j := v1(f)
+					dest(f).SetInt(i + j)
+					return next
+				}
+				return
+			}
+			var valf func(sum int64) reflect.Value
+			switch typ.Kind() {
```
Comment (mvertes) [inline internal/cmd/genop/genop.go:76]: Can you add a TODO here mentioning that this has to be completed with all missing types (int8, ....) ?

File: internal/cmd/genop/genop.go [hunk 2]
```diff
@@ -63,23 +63,73 @@ func {{$name}}(n *node) {
...
 		case c1.rval.IsValid():
 			v0 := genValueInt(c0)
 			{{- if $op.Shift}}
 			j := vUint(c1.rval)
 			{{else}}
 			j := vInt(c1.rval)
 			{{end -}}
+			{{- if (eq $name "add")}}
+			var valf func(sum int64) reflect.Value
+			switch typ.Kind() {
```
Comment (mvertes) [inline internal/cmd/genop/genop.go:105]: idem as line 76: add a TODO for missing types

**Review by mvertes — CHANGES_REQUESTED** [review]

File: internal/cmd/genop/genop.go (hunk 1 above)
Comment (mpl) [reply internal/cmd/genop/genop.go:78]: done

File: internal/cmd/genop/genop.go (hunk 2 above)
Comment (mpl) [reply internal/cmd/genop/genop.go:109]: done

**Review by mvertes — APPROVED** [review]
LGTM

**Pull request merged**